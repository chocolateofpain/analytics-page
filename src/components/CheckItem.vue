<template>
  <table class="table-check-item">
  <tr v-on:click="onClickButton" class="row-height">
    <td class="type-col">
      <Icon :name="showGraph ? 'chevron-down' : 'chevron-right'" :width="12"/>
      <span class="type">{{ checkItemData.type }}</span>
    </td>
    <td>{{ checkItemData.name }}</td>
    <td class="data-col align-center">{{ Math.floor(checkItemData.success * 100) }}%</td>
    <td class="data-col align-center">{{ Math.round(checkItemData.avg) }}ms</td>
    <td class="data-col align-center">{{ Math.round(checkItemData.p95) }}ms</td>
    <td class="data-col align-center">{{ Math.round(checkItemData.p99) }}ms</td>
  </tr>
  <tr v-if="showGraph">
    <td colspan="6">
      <CheckItemChart />
    </td>
  </tr>
  </table>
</template>

<script>
import CheckItemChart from './CheckItemChart.vue'
import Icon from './library/icon/Icon.vue'

export default {
  name: 'CheckItem',
  components: {
    CheckItemChart,
    Icon
  },
  props: [
    'checkItemData'
  ],
  data () {
    return {
      showGraph: false
    }
  },
  methods: {
    onClickButton () {
      this.showGraph = !this.showGraph
    }
  }
}
</script>

<style scoped>
.type {
  margin-left: 4px;
  vertical-align: middle;
}
.table-check-item {
  border-bottom: 1px solid #D2D2D2;
  border-collapse: collapse;
  table-layout: fixed;
  width: 100%;
}
</style>
