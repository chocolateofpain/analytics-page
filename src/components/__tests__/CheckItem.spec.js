import { mount } from '@vue/test-utils'
import CheckItem from '../CheckItem.vue'

describe('CheckItem', () => {
  const checkItemData = {
    id: "bc128a29-953b-44a5-a1f0-e114fca2a22f",
    name: "Check 0",
    type: "API",
    success: 0.5071,
    avg: 500.6750666666667,
    p95: 950,
    p99: 989 
  }
  it('should be defined', () => {
    expect(CheckItem).toBeDefined()
  })

  describe('snapshots', () => {
    it('should render component', () => {
      const component = mount(CheckItem, { propsData: { checkItemData }  })
      expect(component).toMatchSnapshot()
    })
  })
})
