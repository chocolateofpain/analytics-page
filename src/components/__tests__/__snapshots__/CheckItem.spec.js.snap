// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckItem snapshots should render component 1`] = `
<table class="table-check-item">
  <tr class="row-height">
    <td class="type-col">
      <div class="icon"><svg version="1.1" width="12" height="12" viewBox="0 0 16 16" class="octicon octicon-chevron-right" aria-hidden="true">
          <path fill-rule="evenodd" d="M6.22 3.22a.75.75 0 011.06 0l4.25 4.25a.75.75 0 010 1.06l-4.25 4.25a.75.75 0 01-1.06-1.06L9.94 8 6.22 4.28a.75.75 0 010-1.06z"></path>
        </svg></div> <span class="type">API</span>
    </td>
    <td>Check 0</td>
    <td class="data-col align-center">50%</td>
    <td class="data-col align-center">501ms</td>
    <td class="data-col align-center">950ms</td>
    <td class="data-col align-center">989ms</td>
  </tr>
  <!---->
</table>
`;
